WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/RemoteOrderSynchronizer.swift

```diff
@@ -89,6 +89,27 @@ final class RemoteOrderSynchronizer: OrderSynchronizer {
             }
             .store(in: &subscriptions)
     }
+
+    /// Deletes the order if it has been synced remotely.
+    ///
+    func discardOrder() {
+        Just(order)
+            .filter { // Only continue if the order has been synced remotely.
+                $0.orderID != .zero
+            }
+            .sink { order in
+                let action = OrderAction.deleteOrder(siteID: order.siteID, order: order, deletePermanently: true) { result in
+                    switch result {
+                    case .success:
+                        break
+                    case .failure(let error):
+                        DDLogError("⛔️ Error deleting draft order #\(order.orderID): \(error)")
+                    }
+                }
+                self.stores.dispatch(action)
+            }
+            .store(in: &subscriptions)
+    }
```

Comment:
I don't think we need to do the `Just()` + `.sink` dance. We can dispatch the action normally, right? 
Also I'm a bit worried about having this action inside the synchronizer, as someone could call `discardOrder` and keep submitting input triggers.

I'm wondering if we should block inputs somehow, Or maybe I'm just over-worrying?  🤔

Reply:
> We can dispatch the action normally, right?

Sure! I did it this way mainly because we're using Combine in the rest of the synchronizer, and it was interesting to organize it this way, but it isn't really needed.

I went ahead and moved the action out of the synchronizer and back to the new order view model in [2d25276](https://github.com/woocommerce/woocommerce-ios/pull/6420/commits/2d25276abc6167d778000212ad65592d36bd129e), dispatching it normally. I thought about blocking inputs, but right now we immediately exit the new order screen after firing the request (to be more responsive / not block the navigation while waiting for the response) so I don't think there's significant risk if we don't block inputs at this point. What do you think?